src/main.ts

```diff
@@ -2,17 +2,22 @@ import * as cp from 'child_process'
 import * as os from 'os'
 import { AutoLanguageClient, ActiveServer } from 'atom-languageclient'
 import * as path from 'path'
-import { CompositeDisposable, FilesystemChangeEvent } from 'atom';
+import { CompositeDisposable, FilesystemChangeEvent } from 'atom'
 import * as languageServer from 'ocaml-language-server'
 import * as fs from 'fs-extra'
 import merge from 'deepmerge'
 import * as pkg from '../package.json'
 import config from './config'
 import * as Utils from './utils'
-import { DeepPartial, Config } from './types'
+import { DeepPartial, Config, FileExtension } from './types'
 
 const confFile = ".atom/ide-reason.json"
-const defaultConfig = languageServer.ISettings.defaults.reason
+const defaultConfig = merge(
+  {},
+  languageServer.ISettings.defaults.reason,
+  // @ts-ignore: `bsc` doesn't exist in languageServer.ISettings
+  { path: { bsc: 'bsc' } },
+)
```

Comment:
As `bsc` doesn't exist in `languageServer.ISettings` I had to ignore this one. Any suggestions how to handle it better are appreciated.

Reply:
I think you can extends type `Config` with custom fields like this:

```ts
// types.ts
export type OLSConfig = ISettings['reason']
export interface Config extends OLSConfig {
    path: OLSConfig['path'] & { bsc: string }
}
```

then manually pass the generic type to `deepmerge`, thanks for @clouds56 's `DeepPartial` link, it should handle this case perfectly:

```ts
// main.ts
const defaultConfig = merge<Config>(
  languageServer.ISettings.defaults.reason,
  {
    path: {
      bsb: 'bsb'
    }
  }
)
```

ps:`deepmerge` can only merge two object one time, the thrid parameter is options.

Reply:
Works like a charm, thanks!